### PR TITLE
add PutObjectOptions support

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: api/src/main/java/io/minio/PutObjectOptions.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  // allowed minimum part size is 5GiB in multipart upload.
+  public static final long MAX_PART_SIZE = 5L * 1024 * 1024 * 1024;
+  public static final int MAX_MULTIPART_COUNT = 10000;
+
+  private long objectSize = -1;
+  private long partSize = -1;
+  private int partCount = -1;
+  private String contentType = "application/octet-stream";
+  private Map<String, String> headers = null;
+  private ServerSideEncryption sse = null;
```
Comment (kannappanr) [inline api/src/main/java/io/minio/PutObjectOptions.java]: can we also add `legal-hold, `retention` and `tags` as part of this?

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -4763,7 +4727,10 @@ private String putObject(String bucketName, String objectName, Object data, int
    *          Size of object data.
    * @param data
    *          Object data.
+   *
+   * @deprecated As of release 7.0
    */
+  @Deprecated
```
Comment (sinhaashish) [inline api/src/main/java/io/minio/MinioClient.java]: Do we need to deprecate a private function ?

File: api/src/main/java/io/minio/PutObjectOptions.java (hunk 1 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/PutObjectOptions.java]: Ashish will add those.

File: api/src/main/java/io/minio/MinioClient.java (hunk 2 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Yep

File: api/src/main/java/io/minio/PutObjectOptions.java (hunk 1 above)
Comment (kannappanr) [reply api/src/main/java/io/minio/PutObjectOptions.java]: then should that be done before we make a release?

Comment (balamurugana) [reply api/src/main/java/io/minio/PutObjectOptions.java]: Is this PR release blocker?

Comment (sinhaashish) [reply api/src/main/java/io/minio/PutObjectOptions.java]: We need to have this pr in first and then i will add legal hold/ retention and tags

Comment (sinhaashish) [reply api/src/main/java/io/minio/PutObjectOptions.java]: Release can be done once retention pr is merged on top of this pr

**Comment by ZawadaZSE** [comment]
Hey, when will this be released?

**Comment by harshavardhana** [comment]
Soon @ZawadaZSE

**Comment by kathzhan** [comment]
Have the change already be released in some version?

**Comment by ZawadaZSE** [comment]
@harshavardhana Could you be more specific? It's been already over a month.

**Pull request merged**